python_modules/dagster/dagster/_grpc/client.py

```diff
@@ -105,21 +105,56 @@ def _channel(self):
         ) as channel:
             yield channel
 
-    def _query(self, method, request_type, timeout=DEFAULT_GRPC_TIMEOUT, **kwargs):
+    def _get_response(
+        self,
+        method,
+        request,
+        metadata: Optional[List[Tuple[str, str]]] = None,
+        timeout=DEFAULT_GRPC_TIMEOUT,
+    ):
+        with self._channel() as channel:
+            stub = DagsterApiStub(channel)
+            return getattr(stub, method)(request, metadata=metadata, timeout=timeout)
```

Comment:
do you have any idea why we follow this pattern

Reply:
only max knows